### PR TITLE
fix: commands to accept common sdk params

### LINKED ISSUE #730: `akashctl events`: not accepting common sdk params
`akashctl events` doesn't accept `--node`, `--home` which it needs to connect to the right node.

As a workaround, `AKASHCTL_NODE`, `AKASHCTL_HOME` environment variables are honored.

I think that #700 _might_ be the source of this issue.

### PR DESCRIPTION
**Pull request by troian**
fixes #730 issue with viper when setting flags in cobra commands from env

### CODE REVIEW

**Comment by troian** [comment]
looks like it was made to bind `cosmos` flags

**Comment by boz** [comment]
`--home` and things like `--node` all come from cosmos tho.  basically everything but what's in `x/*/client/cli/flags.go` is from cosmos-sdk.

File: cmd/akashctl/main.go [hunk 1]
```diff
@@ -41,8 +40,28 @@ func main() {
...
+
+		if err := viper.BindPFlag(cli.EncodingFlag, root.PersistentFlags().Lookup(cli.EncodingFlag)); err != nil {
+			return err
+		}
+
+		if err := viper.BindPFlag(cli.OutputFlag, root.PersistentFlags().Lookup(cli.OutputFlag)); err != nil {
+			return err
+		}
+
+		if err := viper.BindPFlag(flags.FlagNode, root.Flags().Lookup(flags.FlagNode)); err != nil {
```
Comment (boz) [inline cmd/akashctl/main.go:60]: are all of these flags always present?  `akashctl keys list` for instance might not take `--node`.

It's fine to have here so long as it doesn't crash when `root.Flags().Lookup(flags.FlagNode)` returns `nil`

Comment (troian) [reply cmd/akashctl/main.go]: it does indeed

File: cmd/akashctl/main.go [hunk 2]
```diff
@@ -41,8 +40,25 @@ func main() {
 
 	// Add --chain-id to persistent flags and mark it required
 	root.PersistentFlags().String(flags.FlagChainID, "", "Chain ID of tendermint node")
+	root.PersistentFlags().String(flags.FlagNode, "", "The node address")
+	_ = viper.BindPFlag(flags.FlagNode, root.PersistentFlags().Lookup(flags.FlagNode))
```
Comment (boz) [inline cmd/akashctl/main.go:44]: Why is this different than the others?

FWIW the `--node` param usually comes from `flags.GetCommands()` and `flags.PostCommands()` in `cosmos/cosmos-sdk/client/flags/flags.go`.  See `x/deployment/client/cli/{query,tx}.go`.

The `events` command is not wrapped in that; I think you should just add it in `events/cmd/root.go`.

```go
		c.Flags().String(FlagNode, "tcp://localhost:26657", "<host>:<port> to Tendermint RPC interface for this chain")
		viper.BindPFlag(FlagNode, c.Flags().Lookup(FlagNode))
```

File: events/cmd/root.go [hunk 3]
```diff
@@ -18,6 +20,14 @@ func EventCmd(cdc *codec.Codec) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "events",
 		Short: "Prints out akash events in real time",
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			cmd.Flags().String(flags.FlagNode, "", "The node address")
```
Comment (boz) [inline events/cmd/root.go:24]: i think you want to do this outside of `PreRunE`?

Comment (troian) [reply events/cmd/root.go]: right, that was rather stupid

File: cmd/akashctl/main.go [hunk 4]
```diff
@@ -41,8 +40,23 @@ func main() {
 
 	// Add --chain-id to persistent flags and mark it required
 	root.PersistentFlags().String(flags.FlagChainID, "", "Chain ID of tendermint node")
+
 	root.PersistentPreRunE = func(_ *cobra.Command, _ []string) error {
-		return initConfig(root)
+		// viper bindings below should be applied to root command rather then
+		// to an argument from this function. otherwise viper won't able to find them
```
Comment (Ropes) [inline cmd/akashctl/main.go:46]: That is annoying, but if it works :ship:

File: events/cmd/root.go [hunk 5]
```diff
@@ -25,6 +27,9 @@ func EventCmd(cdc *codec.Codec) *cobra.Command {
 		},
 	}
 
+	cmd.Flags().String(flags.FlagNode, "", "The node address")
+	_ = viper.BindPFlag(flags.FlagNode, cmd.Flags().Lookup(flags.FlagNode))
```
Comment (Ropes) [inline events/cmd/root.go:31]: Curious: Is this FlagNode variable related to the other changes or just a fix while editing things?

Comment (troian) [reply events/cmd/root.go:31]: that ensures flag `node` processed for `events` command

**Pull request merged**